### ISSUE TITLE
Version-stamp saved artifacts (weights, scenarios, demonstrations) with game version + config manifest

### ISSUE THREAD

**Issue by eriiiko** [Apelsinmack/Civulator#28]
Saved artifacts (trained weights, scenarios, future demonstrations, stats files) don't record which version of the game rules produced them. When combat/economy rules change, old demonstration data and old win-rate baselines silently become incomparable — a problem for a project whose method is comparing against baselines. Concrete example: once #24 (ranged hex-distance bug) is fixed, any combat demonstrations recorded before the fix were played under different rules.

**Existing pieces to build on:** `CHANGELOG.md` tracks a semver game version (v0.3.x/v0.4.0), `weights/trained/manifest.md` is a curated human-readable registry, scenario JSONs already store their map seed.

**Approved design (Erik, 2026-08-22 — agreed in chat):**

1. **Single source of truth**: add `__version__ = "0.4.0"` to `civulator/__init__.py`. Bump it together with a CHANGELOG entry whenever *gameplay rules* change (not tooling/viewer changes).
2. **One canonical helper**: new module `civulator/meta.py` with `build_manifest() -> dict` returning:
   ```python
   {
     "game_version": civulator.__version__,
     "git_commit": <short hash via subprocess, "unknown" if git absent>,
     "config": <full CFG dict snapshot>,   # small file, embed it whole — no hash-only
     "date": <ISO 8601 UTC>,
   }
   ```
3. **Embed, don't sidecar** (parallel files drift and get separated):
   - Weights: save `torch.save({"state_dict": ..., "manifest": build_manifest()})`. Loaders accept both the new dict format and legacy bare state_dicts.
   - Scenario / demonstration JSONs: add a top-level `"manifest"` key at save time (scenario_painter.py `save()`, future order recorder).
   - Stats/win-history files: first line = JSON manifest header (or a `_manifest` key if the file is already JSON).
4. `weights/trained/manifest.md` stays as the human-curated registry; embedded manifests are the machine-readable truth.

**Implementation checklist (est. ~half a session):**
- [ ] `civulator/__init__.py`: `__version__`
- [ ] `civulator/meta.py`: `build_manifest()` + unit test (fields present, config matches CFG)
- [ ] `training/trainer.py` `_save_checkpoints` + `save_win_history` + `save_build_stats`: embed manifest
- [ ] Weight-loading sites (`scripts/watch.py`, `scripts/tournament*.py`, `scripts/replay.py`, trainer resume): accept dict-or-bare format via one shared `load_weights()` helper in `civulator/meta.py`
- [ ] `scripts/scenario_painter.py` `save()`: embed manifest
- [ ] Order Recorder (when built, see combat tool design doc Phase 2): embed manifest in demonstration files from day one
- [ ] Add `meta.build_manifest()` to the canonical-systems table in CLAUDE.md ("everyone saving artifacts must use it")
- [ ] CHANGELOG entry

**Explicitly rejected** (decided, don't revisit without new evidence): version numbers inside CLAUDE.md/rules files (rules change rarely; last-updated date suffices), and sidecar `.meta.json` companion files.

**Comment by eriiiko** [comment]
Implementation detail (auto-closed by commit 16a07b1): civulator/meta.py (build_manifest / save_weights / load_weights), __version__=0.4.1, manifests embedded in weights (.pth dict), scenario JSONs, and stats .npy payloads; legacy bare weight files still load (manifest=None). All 14 save/load sites converted; archive/ untouched. CHANGELOG v0.4.1 entry added; canonical-systems table updated.
